bin/varnishd/cache/cache_ws.c

```diff
@@ -104,8 +104,7 @@ WS_Init(struct ws *ws, const char *id, void *space, unsigned len)
 	*ws->e = 0x15;
 	ws->f = ws->s;
 	assert(id[0] & 0x20);
-	assert(strlen(id) < sizeof ws->id);
-	strcpy(ws->id, id);
+	AZ(VSB_strcpy(ws->id, id));
```

Comment:
wouldn't a `bstrcpy()` paralleling the existing `bprintf()` make more sense?

The VSBs are great, but adding the overhead for a single strcpy into a fixed length destination buffer seems not worth it to me.

Reply:
I opened this RFC exactly for this kind of feedback. I'm not worried by the VSB overhead but I was wondering whether someone else would. Also, does it matter whether we use a VSB if it's not performance-sensitive code?

In itself the call site doesn't change much. Since the arguments part of the signature is the same, the only thing you do in most cases is prefix the function with `VSB_` and guard it with `AZ`. I'm, fully aware that `bstrcpy` and friends would require even less changes, but the error handling would become implicit (I tend to prefer explicit error handling).

Reply:
My answer really would be the usual _it depends_: 
* the `bprintf()`-style idea makes sense where the buffer is fixed and there is only one thing to cat/print.
* VSB makes sense where one needs multiple calls and/or a variable size buffer.

So, to summarize, I think the following would make sense:
* ban `strcpy`/`strcat`
* add `bstrcpy`